curriculum/challenges/english/25-front-end-development/lecture-importance-of-accessibility-and-good-html-structure/672a551975938a916c74802c.md

```diff
@@ -10,6 +10,41 @@ dashedName: what-is-the-aria-describedby-attribute
 
 Watch the lecture video and answer the questions below.
 
+# --transcript--
+
+What is the `aria-describedby` attribute, and how does it work?
+
+The `aria-describedby` attribute is used to provide additional information about an element to screen reader users by referencing existing content on the page. It creates a programmatic association between the element and the content (technically referred to as an accessible description), which screen readers can use to inform users of the additional information when they interact with the element. 
+
+The most common use for `aria-describedby` is to associate instructions and error messages with form inputs. Due to the various methods screen reader users have for navigating a page, they may miss these messages when navigating between inputs. Using `aria-describedby` helps ensure that they will hear them.
+
+Let's take a look at a few examples to understand how it works.
+In this first example, we have a form element that accepts a password:
+
+```html
+<form>
+  <label for="password">Password:</label>
+  <input type="password" id="password" aria-describedby="password-help" />
+  <p id="password-help">Your password must be at least 8 characters long.</p>
+</form>
+```
+
+We are using a `label` element for the Password text and associating that with the password `input` field.
+We also have a paragraph element that describes the password requirements. We are using the `aria-describedby` attribute to associate the password `input` field with the password requirements in the paragraph element. When a screen reader user interacts with this `input`, their screen reader will announce the name of the `input` ("Password") and may then also announce the password requirements. This is not an absolute guarantee however as some screen readers may not automatically announce the additional content, or may only announce it in specific circumstances. This is not uncommon. Each screen reader is different and handles ARIA attributes in their own way. This does not negate the use of `aria-describedby`, as it will benefit screen reader users in general.
```

Comment:
```suggestion
We also have a paragraph element that describes the password requirements. We are using the `aria-describedby` attribute to associate the password `input` field with the password requirements in the paragraph element. When a screen reader user interacts with this `input`, their screen reader will announce the name of the `input` (`"Password"`) and may then also announce the password requirements. This is not an absolute guarantee however as some screen readers may not automatically announce the additional content, or may only announce it in specific circumstances. This is not uncommon. Each screen reader is different and handles ARIA attributes in their own way. This does not negate the use of `aria-describedby`, as it will benefit screen reader users in general.
```